lib/ansible/modules/database/mysql/mysql_db.py

```diff
@@ -222,16 +221,12 @@ def db_import(module, host, user, password, db_name, target, all_databases, port
         comp_prog_path = module.get_bin_path('xz', required=True)
 
     if comp_prog_path:
-        p1 = subprocess.Popen([comp_prog_path, '-dc', target], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        p2 = subprocess.Popen(cmd, stdin=p1.stdout, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        (stdout2, stderr2) = p2.communicate()
-        p1.stdout.close()
-        p1.wait()
-        if p1.returncode != 0:
-            stderr1 = p1.stderr.read()
-            return p1.returncode, '', stderr1
+        rc1, stdout1, stderr1 = module.run_command([comp_prog_path, '-dc', target])
+        if rc1 != 0:
+            return rc1, '', stderr1
         else:
-            return p2.returncode, stdout2, stderr2
+            rc2, stdout2, stderr2 = module.run_command(cmd, data=stdout1, binary_data=True)
+            return rc2, stdout2, stderr2
```

Comment:
The change to this module is the only one that I question.  I'm leaning towards putting this on the allowed list.  The reason is that the code presently in the module is the equivalent of a UNIX pipe.  The decompressor is able to chunk the data from the file to the database program as it decompresses that portion of the file.  The run_command() version has to store all of the data from the decompressor in memory before passing it to the database program.  So the memory usage can balloon in this case.